### ISSUE TITLE
Testing Unit Tests Parsing with 0.9.4-RC1 - repeated log messages

### ISSUE THREAD

**Issue by Benoitlz** [wenns/sonar-cxx#535]
There is a new issue in log file now: each syntax error generates 20 to 40 times the same log warning, e.g.:
19:45:14.637 WARN - [M:\wcs_simulation_QA_385...\FluidDensityInterfaceTest.cpp:115]: syntax error, skip 'testing'

As I have some issues parsing Google Tests, sonar_runner generates a 1.9 GB log file!

The way it was reported with 0.9.3 is ok, although more information would be welcome.
With 0.9.4-RC1 the same log line is repeated 20 to 40 time ... should be output only once!
Thanks.

**Comment by guwirth** [comment]
This seems to be a side effect of https://github.com/wenns/sonar-cxx/commit/f99b5dc2527bde34f8f662f9fecd250364064004. The scanner adds the nodes multiple times to the AST. 

Is there a possibility to clear the AST without rebuilding the scanner?

**Comment by guwirth** [comment]
Seems we have to revert the initial change #524 as long there is no solution to reset the parse/scanner?
